modules/influxdb/src/test/java/org/testcontainers/containers/InfluxDBContainerTest.java

```diff
@@ -1,44 +1,137 @@
+
+    private static final String BUCKET = "new-test-bucket";
+
+    private static final String RETENTION = "1w";
+
+    private static final String ADMIN_TOKEN = "super-secret-token";
+
+    @Nullable
+    private InfluxDBClient influxDBClient = null;
+
+    @After
+    public void stopInfluxDBClient() {
+        if (this.influxDBClient != null) {
+            this.influxDBClient.close();
+            this.influxDBClient = null;
+        }
+    }
 
     @Test
     public void getUrl() {
-        String actual = influxDBContainer.getUrl();
+        try (
+            final InfluxDBContainer influxDBContainer = new InfluxDBContainer(InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
+        ) {
+            influxDBContainer.start();
+            assertThat(influxDBContainer.isRunning()).isTrue();
 
-        assertThat(actual).isNotNull();
+            final String actual = influxDBContainer.getUrl();
+            assertThat(actual).isNotNull();
+        }
     }
 
     @Test
-    public void getNewInfluxDB() {
-        InfluxDB actual = influxDBContainer.getNewInfluxDB();
+    public void getInfluxDBClient() {
+        try (
+            final InfluxDBContainer influxDBContainer = new InfluxDBContainer(InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
+        ) {
+            influxDBContainer.start();
+            assertThat(influxDBContainer.isRunning()).isTrue();
+
+            this.influxDBClient = InfluxDBTestUtils.getInfluxDBClient(influxDBContainer);
 
-        assertThat(actual).isNotNull();
-        assertThat(actual.ping()).isNotNull();
+            assertThat(this.influxDBClient).isNotNull();
+            assertThat(this.influxDBClient.ping()).isTrue();
+        }
     }
 
     @Test
-    public void getLivenessCheckPort() {
-        Integer actual = influxDBContainer.getLivenessCheckPort();
+    public void getBucket() {
+        try (
+            final InfluxDBContainer influxDBContainer = new InfluxDBContainer(InfluxDBTestUtils.INFLUXDB_V2_TEST_IMAGE)
+        ) {
+            influxDBContainer
+                .withUsername(USERNAME)
+                .withPassword(PASSWORD)
+                .withOrganization(ORG)
+                .withBucket(BUCKET)
+                .withRetention(RETENTION)
+                .withAdminToken(ADMIN_TOKEN);
 
-        assertThat(actual).isNotNull();
+            influxDBContainer.start();
+            assertThat(influxDBContainer.isRunning()).isTrue();
+
+            this.influxDBClient = InfluxDBTestUtils.getInfluxDBClient(influxDBContainer);
+
+            assertThat(this.influxDBClient).isNotNull();
+
+            final Bucket bucket = this.influxDBClient.getBucketsApi().findBucketByName(BUCKET);
+            assertThat(bucket).isNotNull();
+
+            assertThat(bucket.getName()).isEqualTo(BUCKET);
+        }
     }
 
     @Test
-    public void isRunning() {
-        boolean actual = influxDBContainer.isRunning();
+    public void queryForWriteAndRead() {
```

Comment:
I don't understand the idea of this test. Is there anything in the implementation code that would make this test fail, but would still keep all other tests green?

Reply:
IMO this test ensures that the written values are read correctly and all the setup is done right.

Reply:
I would keep it